### PR TITLE
[REG-1099] Documentation for GitHub Actions config

### PR DESCRIPTION
**Pull request by vontell**
Docs for the GitHub Actions configuration of our docs. Not fully tested because I am fixing one part of the yaml atm, but it is ready for a review!

### CODE REVIEW

**Review by rcornwall-reg — APPROVED** [review]

File: docs/studios/unity/tutorials/github_actions.md [hunk 1]
```diff
@@ -0,0 +1,309 @@
+---
+sidebar_label: 'Testing with GitHub Actions'
+---
+
+# Setting up QA Bots with GitHub Actions
+
+This tutorial will walk you through setting up a QA Bot for your Unity project using GitHub Actions. This tutorial assumes you have
+some familiarity with [GitHub Actions](https://docs.github.com/en/actions), and that you have a working bot within your
+game (see more on this within the [Creating your first bot guide](/studios/unity/unity-sdk/creating-bots/configuration)). 
+You can find a working example in [this repository](google.com) for reference.
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:10]: google.com ?

File: docs/studios/unity/tutorials/github_actions.md [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+This guide and reference is subject to change, as the product is in an **early preview**. We
+are making changes based on your recommendations and feedback, so let us know what you think at
+info@regression.gg, or join our [Discord](https://discord.com/invite/925SYVse2H).
+
+:::
+
+:::info
+
+This features requires a license for Unity, since CI/CD builds for Unity are only available if you have
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:22]: I'll let @abeizer suggest new wordings for these sentences as she is good at that, but they read a bit awkwardly as currently written.

File: docs/studios/unity/tutorials/github_actions.md [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+a license for Unity. Additionally, if you want to use your existing CI/CD pipelines, the best reference
+here is the environment variable and test function sections below.
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:24]: These should be links

File: docs/studios/unity/tutorials/github_actions.md [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+:::
+
+## Overview
+
+In this guide, we will use GitHub Actions, GameCI, and Regression Games to set up a QA Bot for your Unity project. Whenever a new
+push is made within your project, the Unity project will be built and a bot will be started. The steps to this
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:31]: `within your project`, should probably be `to your GitHub repository`

File: docs/studios/unity/tutorials/github_actions.md [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+1. Add the GitHub Actions workflow files to your project needed to get Unity builds to work.
+2. Configuring the GitHub Actions workflow files to work with GameCI and Regression Games.
+3. Add a test to your Unity project that will start and run your bot.
+4. Learn more about how to view test results.
+
+## Add GitHub Action Workflow Files
+
+We use the [GameCI](https://game.ci/) project as a foundation for running GitHub CI/CD builds for Unity. The GameCI project provides a collection of GitHub Actions that are used to build Unity projects.
+
+#### 1. Create a new branch in your project called `qa-tests`.
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:43]: is this branch-name required ?  What is the significance of this specific name.  We should explain that to them and/or let them know that they can use their own branch name and what other settings need to adjust to accommodate it.

File: docs/studios/unity/tutorials/github_actions.md [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+
+#### 2. Follow the steps on GameCI to activate a [Unity License](https://game.ci/docs/github/activation).
+
+It is critical that you follow this step! This allows the CI/CD Docker images to properly start and use Unity.
+You can test that the license is configured properly by running this [GitHub Action](https://github.com/marketplace/actions/unity-activate)
+in your project.
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:49]: Clarification question... In my project or in my Github repository's actions ?

File: docs/studios/unity/tutorials/github_actions.md [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+
+#### 3. Add the QA Bot workflow file to your project.
+
+The QA Bot workflow file is a GitHub Action that will build your Unity project and start a test that will run a bot. From step 2 above,
+you should have a `.github/workflows` folder. Inside of this folder, create a new file called `qa-bot.yml` and paste the contents below.
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:54]: I should have this folder where ?

In my Github repository ?
In my unity project ?

Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:54]: I feel like they should be able to download this file as an alternative to cut/paste contents

File: docs/studios/unity/tutorials/github_actions.md [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+Note that this will actually run all of your tests, not just the QA Bot test. To configure this workflow to only
+run the QA test, see the [Unity Test Runner Command Line docs](https://docs.unity3d.com/Packages/com.unity.test-framework@2.0/manual/reference-command-line.html),
+which has filter options that can be passed within `customParameters` within the `Run tests` step above.
+
+:::
+
+## Configure GitHub Actions
+
+The final piece on the GitHub actions side is to configure the secrets that are used in the workflow file above. Visit the secrets
+page for your project by going to your **Repo** > **Settings** > **Secrets and variables** > **Actions**. Add the following secrets
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:128]: unity project or github repository ?

File: docs/studios/unity/tutorials/github_actions.md [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+by clicking the green **New repository secret** button.
+
+1. `RG_HOST` - `https://play.regression.gg`
+2. `RG_API_KEY` - Your API key from Regression Games, which can be found in your [account settings](https://play.regression.gg/account).
+                  Full instructions for this can be found in the [Authentication docs.](studios/api_reference/authentication).
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:133]: extra `.` in the brackets

File: docs/studios/unity/tutorials/github_actions.md [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+
+![Secrets for GitHub Actions](./github_actions_images/secrets.png)
+
+## Add a Test to Your Unity Project
+
+The QA Bots are started via a the [Unity Test Runner as a Play Mode test](https://docs.unity3d.com/2019.1/Documentation/Manual/PlaymodeTestFramework.html). 
+In this testing mode, the game starts as if you are playing it, and then terminates once the function finishes.
+
+Complete the following steps to setup the required test - feel free to skip some of these steps if you have existing Play tests
+in your project.
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:149]: project or repository ?  I'm making this comment a lot, but this guide needs to provide more context of which thing I'm working in for each section.  Unity or Github or ??? before telling me a set of steps to take.

File: docs/studios/unity/tutorials/github_actions.md [hunk 11]
```diff
@@ -0,0 +1,309 @@
...
+Before adding the test, make sure your desired test scenes are placed inside your build. You can add your scenes to the build
+using the menu option **File** > **Build Settings**.
+
+![Build settings](./github_actions_images/build.png)
+
+#### 2. Configure Test Runner for your project
+
+Next, we need to configure the Unity Test Runner for your project. To do this, go to **Window** > **General** > **Test Runner**.
+Then, in the pane that open, click **Play Mode** > **Create PlayMode Test Assembly Folder**. Create that tests folder with
+the name `Tests`, and then open that folder in your editor. You should see a file called `Tests.asmdef` - open that file and
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:162]: which editor ?  I don't generally see the term editor used when opening a folder, normally more for a file.

File: docs/studios/unity/tutorials/github_actions.md [hunk 12]
```diff
@@ -0,0 +1,309 @@
...
+#### 1. Add your scenes to your build
+
+Before adding the test, make sure your desired test scenes are placed inside your build. You can add your scenes to the build
+using the menu option **File** > **Build Settings**.
+
+![Build settings](./github_actions_images/build.png)
+
+#### 2. Configure Test Runner for your project
+
+Next, we need to configure the Unity Test Runner for your project. To do this, go to **Window** > **General** > **Test Runner**.
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:160]: I feel like we should have our own RG specific extension menu option here that does this step for them. (The creating the asmdef file parts)

File: docs/studios/unity/tutorials/github_actions.md [hunk 13]
```diff
@@ -0,0 +1,309 @@
...
+        "TestAssemblies"
+    ]
+}
+```
+
+You can learn more about the Test Runner and setup instructions [here](https://docs.unity.cn/560/Documentation/Manual/testing-editortestsrunner.html).
+
+#### 3. Create an `RGBotTests` file to run your bot
+
+Within the `Tests` folder, create a new file called `RGBotTests.cs` and copy the following contents into the file.
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:181]: again, we should give them the option to download the file as an alternative to cut/paste contents

File: docs/studios/unity/tutorials/github_actions.md [hunk 14]
```diff
@@ -0,0 +1,309 @@
...
+
+:::caution
+
+Make sure to edit the file below to use the correct starting scene for your test! Update `Scenes/SampleScene` to
+instead be the scene you are starting with.
+
+Also, update `defaultBotId` to an ID of a bot that you'd like to start in your local editor. Remember, you can find
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:188]: This second part feels less like a 'caution' and more like an 'optional/suggested' thing

File: docs/studios/unity/tutorials/github_actions.md [hunk 15]
```diff
@@ -0,0 +1,309 @@
...
+    
+}
+
+```
+
+At a high level, this test is doing the following:
+
+1. Loading the scene, and waiting for it to start up before activating bots
+2. Queuing up a task that will start a bot and spawn it into the scene
+3. The test then waits for at least one bot to connects. If a bot does not connect within one minute, the test will fail.
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:271]: `to connects` -> `to connect`

File: docs/studios/unity/tutorials/github_actions.md [hunk 16]
```diff
@@ -0,0 +1,309 @@
...
+4. Once at least one bot is connected, the test runs until all bots complete their tasks (i.e the bot calls `rg.complete()`)
+
+#### 4. Test it out with an existing bot
+
+Our test is now setup! Before we commit and push our changes, let's test it out locally with an existing bot. To do this, you can
+follow the [first bot guide](/studios/unity/tutorials/first_tutorial) or [creating bots reference](/studios/unity/unity-sdk/creating-bots/configuration)
+to create and configure a bot. As a quick test, you can make a bot that simply waits for a few seconds before completing it's test. Note
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:278]: `it's` -> `its`

File: docs/studios/unity/tutorials/github_actions.md [hunk 17]
```diff
@@ -0,0 +1,309 @@
...
+  }
+
+}
+```
+
+## Viewing Test Results
+
+Once a test is done running, you can see your most recent test results within the Regression Games website, within the **History** tab
+of the [Running Bots](https://play.regression.gg/running-bots) page. Within this page, you can download the logs for the run, as well
+as the replay file. Follow the instructions at [here](/studios/unity/unity-sdk/in-editor-replay) for viewing this replay in your local
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/tutorials/github_actions.md:304]: Not a comment to resolve in this PR, but I feel like we should now fill in the validation results column on the UI with # of validations passed and failed... colored accordingly based on 100% pass or not.

**Review by RG-nAmKcAz — CHANGES_REQUESTED** [review]
This seems like a lot of steps that could go wrong or have a cut/paste or typo error break things.

Most of my comments are around giving the user more context of which application the instruction section they're in refers to, and potentially less error prone ways for them to get the needed starter files.

File: docs/studios/unity/tutorials/github_actions.md [hunk 18]
```diff
@@ -0,0 +1,309 @@
...
+      - uses: actions/upload-artifact@v3
+        with:
+          name: Build
+          path: build
+```
+
+:::info
+
+Note that this will actually run all of your tests, not just the QA Bot test. To configure this workflow to only
+run the QA test, see the [Unity Test Runner Command Line docs](https://docs.unity3d.com/Packages/com.unity.test-framework@2.0/manual/reference-command-line.html),
```
Comment (abeizer) [inline docs/studios/unity/tutorials/github_actions.md:120]: In the interest of making setup as quick as possible, it'd be nice to have an example of how you'd do this for a hypothetical project. Ideally something where I'd substitute <my bot test directory> or <my other test directories>

File: docs/studios/unity/tutorials/github_actions.md [hunk 19]
```diff
@@ -0,0 +1,309 @@
...
+## Configure GitHub Actions
+
+The final piece on the GitHub actions side is to configure the secrets that are used in the workflow file above. Visit the secrets
+page for your project by going to your **Repo** > **Settings** > **Secrets and variables** > **Actions**. Add the following secrets
+by clicking the green **New repository secret** button.
+
+1. `RG_HOST` - `https://play.regression.gg`
+2. `RG_API_KEY` - Your API key from Regression Games, which can be found in your [account settings](https://play.regression.gg/account).
+                  Full instructions for this can be found in the [Authentication docs.](studios/api_reference/authentication).
+3. `RG_BOT` - The ID of the bot you want to run. Within the [Bot Manager page](https://play.regression.gg/bots), you can see the ID
```
Comment (abeizer) [inline docs/studios/unity/tutorials/github_actions.md:134]: This reads a bit awkwardly. Suggest rewording to something like "The ID of the bot you want to run. You can find this through the bot's details within the Regression Games Bot Manager." (then the image clarifies the _where_)

File: docs/studios/unity/tutorials/github_actions.md [hunk 20]
```diff
@@ -0,0 +1,309 @@
...
+              for a bot in its details below the name of the bot.
+
+![Bot ID](./github_actions_images/bot_id.png)
```
Comment (abeizer) [inline docs/studios/unity/tutorials/github_actions.md:137]: Not a _huge_ deal but all of our other images use white arrows to make them more obvious against the dark theme

File: docs/studios/unity/tutorials/github_actions.md [hunk 21]
```diff
@@ -0,0 +1,309 @@
...
+
+Once you are done adding these secrets, your secrets section should look similar to the following:
+
+![Secrets for GitHub Actions](./github_actions_images/secrets.png)
+
+## Add a Test to Your Unity Project
+
+The QA Bots are started via a the [Unity Test Runner as a Play Mode test](https://docs.unity3d.com/2019.1/Documentation/Manual/PlaymodeTestFramework.html).
```
Comment (abeizer) [inline docs/studios/unity/tutorials/github_actions.md:145]: This is confusing. Would it be valid to reword this to "The QA Bots are started via the Unity Test Runner in Play Mode" ?

File: docs/studios/unity/tutorials/github_actions.md [hunk 22]
```diff
@@ -0,0 +1,309 @@
...
+
+## Add a Test to Your Unity Project
+
+The QA Bots are started via a the [Unity Test Runner as a Play Mode test](https://docs.unity3d.com/2019.1/Documentation/Manual/PlaymodeTestFramework.html). 
+In this testing mode, the game starts as if you are playing it, and then terminates once the function finishes.
+
+Complete the following steps to setup the required test - feel free to skip some of these steps if you have existing Play tests
+in your project.
+
+#### 1. Add your scenes to your build
```
Comment (abeizer) [inline docs/studios/unity/tutorials/github_actions.md:151]: Same comment as other section about ### and no numbers

File: docs/studios/unity/tutorials/github_actions.md [hunk 23]
```diff
@@ -0,0 +1,314 @@
+---
+sidebar_label: 'Testing with GitHub Actions'
+---
+
+# Setting up a bot with GitHub Actions for quality assurance
```
Comment (abeizer) [inline docs/studios/unity/tutorials/github_actions.md:5]: Make sure your titles are properly capitalized

File: docs/studios/unity/tutorials/github_actions.md [hunk 24]
```diff
@@ -0,0 +1,309 @@
+---
+sidebar_label: 'Testing with GitHub Actions'
+---
+
+# Setting up QA Bots with GitHub Actions
+
+This tutorial will walk you through setting up a QA Bot for your Unity project using GitHub Actions. This tutorial assumes you have
```
[the remark this answers is not in the document]
Comment (abeizer) [reply docs/studios/unity/tutorials/github_actions.md]: Agree on the title, I think it'd make sense to update any references to "QA Bot" within the guide to just "bot" since realistically this could be _any_ kind of bot for now (validation, playtest, ones we create in the future)

File: docs/studios/unity/tutorials/github_actions.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (abeizer) [reply docs/studios/unity/tutorials/github_actions.md]: You'll want to be explicit about that in the guide. Right now it reads as a reference to a Javascript bot, not a project with GitHub Actions - and the README for that project doesn't mention anything about GitHub Actions either, so if I didn't already know what that project does I'd probably be a bit lost.

File: docs/studios/unity/tutorials/github_actions.md (hunk 13 above)
Comment (abeizer) [reply docs/studios/unity/tutorials/github_actions.md:186]: Is this something we could include in our Unity Package? Maybe they'd just need to set set some variables to configure it for their project but they wouldn't need to go download or do any cut/paste?

File: docs/studios/unity/tutorials/github_actions.md [hunk 25]
```diff
@@ -0,0 +1,314 @@
...
+    // Now try to finish
+    rg.complete();
+  }
+
+}
+```
+
+## Viewing Test Results
+
+Once a test is done running, you can see your most recent test results within the Regression Games website, within the **History** tab
```
Comment (abeizer) [inline docs/studios/unity/tutorials/github_actions.md:307]: Suggest reword (sorry this is so different from what you had but it evolved as I wrote 😆 ): 

"Once your tests have completed, you can view their results on Regression Games by navigating to the History section of the Running Bots tab. From this page you can download bot logs for the test session which contain output from your bot-code. You can also download in-game replay data which can be loaded in the Unity Editor to watch the movements and actions of each bot, as well as analyze the validations they performed. See the [In-Editor Replay](/studios/unity/unity-sdk/in-editor-replay) for instructions on how to load replay data."

**Review by abeizer — CHANGES_REQUESTED** [review]

File: docs/studios/unity/tutorials/github_actions.md (hunk 12 above)
Comment (RG-nAmKcAz) [reply docs/studios/unity/tutorials/github_actions.md:165]: does this task exist ? If so I'll resolve this comment

Comment (vontell) [reply docs/studios/unity/tutorials/github_actions.md:165]: Added here: https://linear.app/regression-games/issue/REG-1240/create-menu-option-in-unity-to-automatically-setup-test-files

File: docs/studios/unity/tutorials/github_actions.md (hunk 25 above)
Comment (vontell) [reply docs/studios/unity/tutorials/github_actions.md]: Fixed!

**Review by vontell — COMMENTED** [review]
Addressed your comments @abeizer!

File: docs/studios/unity/tutorials/github_actions.md [hunk 26]
```diff
@@ -0,0 +1,406 @@
...
+[test files](#add-a-test-to-your-unity-project).
+
+:::
+
+## Add GitHub Actions Workflow Files
+
+[GameCI](https://game.ci/) provides a collection of GitHub Actions for building Unity projects. This is our foundation 
+for running GitHub CI/CD builds for Unity.
+
+### Create a New Branch in your Project
```
Comment (abeizer) [inline docs/studios/unity/tutorials/github_actions.md:43]:
```suggestion
### Create a New Branch in Your Project
```

File: docs/studios/unity/tutorials/github_actions.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (abeizer) [reply docs/studios/unity/tutorials/github_actions.md]: This still needs to be addressed

File: docs/studios/unity/tutorials/github_actions.md (hunk 24 above)
[the remark this answers is not in the document]
Comment (abeizer) [reply docs/studios/unity/tutorials/github_actions.md]: I still see references to "QA Bot". Are we planning on keeping this term?

File: docs/studios/unity/tutorials/github_actions.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (vontell) [reply docs/studios/unity/tutorials/github_actions.md]: Oops! Will fix today

**Review by RG-nAmKcAz — APPROVED** [review]

File: docs/studios/unity/tutorials/github_actions.md (hunk 24 above)
[the remark this answers is not in the document]
Comment (vontell) [reply docs/studios/unity/tutorials/github_actions.md]: removed most of these except for where I talk specifically about this use cases in the guide!

**Review by abeizer — APPROVED** [review]
